contracts/.gitignore

```diff
@@ -144,3 +144,7 @@ dist
 .pnp.*
 
 local_rpc.json
+
+# TypeChain files
+/typechain
+/typechain-types
```

Comment:
it's suggested that these files should not be checked-out (since they can be generated)